includes/service-providers/constant_contact/class-newspack-newsletters-constant-contact.php

```diff
@@ -374,45 +374,50 @@ public function save( $post_id, $post, $update ) {
 	/**
 	 * Send a campaign.
 	 *
-	 * @param integer $post_id Post ID to send.
+	 * @param string  $new_status New status of the post.
+	 * @param string  $old_status Old status of the post.
 	 * @param WP_POST $post Post to send.
 	 */
-	public function send( $post_id, $post ) {
+	public function send( $new_status, $old_status, $post ) {
+		$post_id = $post->ID;
+
 		if ( ! Newspack_Newsletters::validate_newsletter_id( $post_id ) ) {
 			return new WP_Error(
 				'newspack_newsletters_incorrect_post_type',
 				__( 'Post is not a Newsletter.', 'newspack-newsletters' )
 			);
 		}
 
-		try {
-			$sync_result = $this->sync( $post );
+		if ( 'publish' === $new_status && 'publish' !== $old_status ) {
```

Comment:
Isn't that essentially replicating the `publish_` action?

Reply:
It is, but the `publish` action gets triggered anytime a published post is updated, too, so I was running into errors when updating meta for newsletters that I had already sent (the action would try to re-send the newsletters every time I updated the post). So using `transition_post_status` and only triggering the send if the post is transitioning from unpublished to published fixed that and also allows the post to be updated after being sent.

Reply:
Got it, thanks for the clarification!